### PR TITLE
Database-as-a-Service Configuration

### PR DESCRIPTION
**Pull request by ytsarev**
### Description of your changes

Initial layout + DBaas Configuration

 * Single XRD to drive the creation of multi-cloud managed database instances
 * AWS and Azure support
 * Uptest

Signed-off-by: Yury Tsarev <yury@upbound.io>

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

```
make e2e
...
--- PASS: kuttl (2228.11s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/case (2227.62s)
PASS
03:54:33 [ OK ] running automated tests
```

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  issue_comment:
+    types: [created]
+
+jobs:
+  e2e:
+    uses: upbound/uptest/.github/workflows/pr-comment-trigger.yml@main
+    with:
+      package-type: configuration
+    secrets:
+      UPTEST_CLOUD_CREDENTIALS: ${{ secrets.UPTEST_CLOUD_CREDENTIALS }}
```
Comment (haarchri) [inline .github/workflows/e2e.yaml:13]: We need Support for Azure and AWS - think you can't use the generic e2e template here - in caas we have a version for multicloud

Comment (ytsarev) [reply .github/workflows/e2e.yaml:13]: Thanks for the great catch 👍 I switched to combined cloud credential format in https://github.com/upbound/configuration-dbaas/pull/1/commits/787a892d26cba93de7bd2d4a8d90e235c577aab1 to comply with the reusable workflow. This way we can avoid GHA workflow duplication and reuse the standard uptest one. 

```
--- PASS: kuttl (1224.85s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/case (1224.35s)
PASS
12:01:21 [ OK ] running automated tests
```

File: examples/postgres-azure-claim.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+kind: SQLInstance
+metadata:
+  name: configuration-dbaas-azure-postgresql
+  namespace: default
+spec:
+  compositionSelector:
+    matchLabels:
+      provider: azure
+  parameters:
+    region: westus
```
Comment (haarchri) [inline examples/postgres-azure-claim.yaml:11]: i wonder if it makes sense to unify the regions for the cloud-providers

**Review by haarchri — APPROVED** [review]
LGTM - one question not blocking

File: examples/postgres-azure-claim.yaml (hunk 2 above)
Comment (ytsarev) [reply examples/postgres-azure-claim.yaml:11]: I generally like the idea but the region layout is deviating per cloud provider so we should probably postpone that decision until GCP mix in and then try to somehow unify all 3. On the other hand, we might limit end users to target specific location. E.g. @elohmrow might like [Switzerland North](https://azure.microsoft.com/de-de/updates/generally-available-switzerland-north-availability-zones/) in Azure and average unification might limit it :)  Let's attack this problem separately with the next evolution of this Configuration.

**Pull request merged**